### ISSUE TITLE
🐛 BUG: same name for the queue and its binding generates internal server error (10002)

### ISSUE THREAD

**Issue by kika** [cloudflare/workers-sdk#2893]
### Which Cloudflare product(s) does this pertain to?

Queues

### What version of `Wrangler` are you using?

2.12.2

### What operating system are you using?

Mac

### Describe the Bug

```
[[queues.producers]]
  queue = "data"
  binding = "data"
```

produces
```
✘ [ERROR] A request to the Cloudflare API (/accounts/xxxxxxx/workers/scripts/zzzzzz) failed.

  workers.api.error.internal_server [code: 10002]
```

renaming binding to anything else fixes it.

**Comment by jspspike** [comment]
I don't believe bindings should be able to use the same name, I will create an internal ticket to improve the api error response and leave this issue open as an enhancement to validate binding names

**Comment by biodranik** [comment]
We hit the same issue after migrating from Wrangler 2.19.0 to 3.1.1, without any bindings in the [wrangler.toml](https://github.com/organicmaps/url-processor/blob/master/wrangler.toml) (but our worker uses static website assets):
https://github.com/organicmaps/url-processor/actions/runs/5466497130/jobs/9951505713

**Comment by gafitescu** [comment]
On my end the problem was that I assign the KV to worker in Cloudflare as well as on the wrangler.toml
